lib/src/internet_connection.dart

```diff
@@ -70,12 +76,18 @@ class InternetConnection {
   ///
   /// - If [useDefaultOptions] is `false`, you must provide a non-empty
   /// [customCheckOptions] list.
+  ///
+  /// The [customConnectivityCheck] allows you to provide a custom method for checking
+  /// endpoint reachability. If provided, it will be used for all connectivity checks
+  /// instead of the default HTTP HEAD request implementation.
   InternetConnection.createInstance({
     Duration? checkInterval,
     List<InternetCheckOption>? customCheckOptions,
     bool useDefaultOptions = true,
     this.enableStrictCheck = false,
+    ConnectivityCheckCallback? customConnectivityCheck,
   })  : _checkInterval = checkInterval ?? _defaultCheckInterval,
+        _reachabilityChecker = customConnectivityCheck,
```

Comment:
This could directly be `this.customConnectivityCheck` cuz we are not doing any modifications.